lib-cp-connector/pkg/api/internal.go

```diff
@@ -0,0 +1,220 @@
+package api
+
+import (
+	"crypto/tls"
+	"github.com/benbjohnson/clock"
+	"github.com/keptn/go-utils/pkg/api/models"
+	api "github.com/keptn/go-utils/pkg/api/utils"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"net/http"
+	"time"
+)
+
+// InternalAPISet is an implementation of APISet
+// which can be used from within the Keptn control plane
+type InternalAPISet struct {
+	apimap                 InClusterAPIMappings
+	httpClient             *http.Client
+	apiHandler             *api.APIHandler
+	authHandler            *api.AuthHandler
+	eventHandler           *api.EventHandler
+	logHandler             *api.LogHandler
+	projectHandler         *api.ProjectHandler
+	resourceHandler        *api.ResourceHandler
+	secretHandler          *api.SecretHandler
+	sequenceControlHandler *api.SequenceControlHandler
+	serviceHandler         *api.ServiceHandler
+	stageHandler           *api.StageHandler
+	uniformHandler         *api.UniformHandler
+	shipyardControlHandler *api.ShipyardControllerHandler
+}
+
+// InternalService is used to enumerate internal Keptn services
+type InternalService int
+
+const (
+	ConfigurationService InternalService = iota
+	ShipyardController
+	ApiService
+	SecretService
+	MongoDBDatastore
+)
+
+// InClusterAPIMappings maps a keptn service name to its reachable domain name
+type InClusterAPIMappings map[InternalService]string
+
+// DefaultInClusterAPIMappings gives you the default InClusterAPIMappings
+var DefaultInClusterAPIMappings = InClusterAPIMappings{
+	ConfigurationService: "configuration-service:8080",
+	ShipyardController:   "shipyard-controller:8080",
+	ApiService:           "api-service:8080",
+	SecretService:        "secret-service:8080",
+	MongoDBDatastore:     "mongodb-datastore:8080",
+}
+
+// NewInternal creates a new InternalAPISet usable for calling keptn services from within the control plane
+func NewInternal(client *http.Client, apiMappings ...InClusterAPIMappings) (*InternalAPISet, error) {
```

Comment:
given that the package only contains this component, should we rename this to just `New()`? Or will this eventually also include an `ExternalAPISet`?

Reply:
It *could* be that there will be a `NewExternal` function besides this one which will just use the regular `APISet` provided by `go-utils` https://github.com/keptn/go-utils/blob/24667368a1594c17cfa725b88a97d19ddfef109e/pkg/api/utils/client.go#L146

Reply:
Update: agreed with @bacherfl  to keep `NewInternal` for now